dot/peerset/handler.go

```diff
@@ -30,67 +34,89 @@ func NewPeerSetHandler(cfg *ConfigSet) (*Handler, error) {
 	}, nil
 }
 
+func (h *Handler) setActionQueue(act action) {
+	go func(data action) {
+		h.writerWGMutex.Lock()
+		h.writersWG.Add(1)
+		h.writerWGMutex.Unlock()
+
+		defer h.writersWG.Done()
+
+		select {
```

Comment:
I'm a bit confused what the purpose of these two selects are, why cant they just be one? Would you mind explaining just for my own learning?

Reply:
basically, the first one prevents later goroutines to write on the already closed channel, the second is when the close case is triggered while we are blocked trying to write in a full channel, but this code is a way to gracefully close a channel that is being used by other goroutines (since closing a channel while writing causes race condition).

Reply:
Ooofff this bad. You don't have any guarantee `h.closeCh` won't get closed between the two select blocks (and hence closing `h.actionQueue`). Please change somehow.